modules/core/src/main/scala/lucuma/odb/api/model/SpectroscopyRequirements.scala

```diff
@@ -0,0 +1,133 @@
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import io.circe.refined._
+import lucuma.odb.api.model.FocalPlane
+import lucuma.odb.api.model.SpectroscopyCapabilities
+import lucuma.core.math.Angle
+import lucuma.core.util.Display
+import lucuma.core.math.Wavelength
+import lucuma.core.util.Enumerated
+
+final case class SpectroscopyScienceRequirements(
+  wavelength:       Option[Wavelength],
+  resolution:       Option[PosInt],
+  signalToNoise:    Option[PosBigDecimal],
+  signalToNoiseAt:  Option[Wavelength],
+  wavelengthRange:  Option[Wavelength],
+  focalPlane:       Option[FocalPlane],
+  focalPlaneAngle:  Option[Angle],
+  capabilities:     Option[SpectroscopyCapabilities]
+)
+
+object SpectroscopyScienceRequirements {
+  val Default = SpectroscopyScienceRequirements(None, refineMV[Positive](1).some, None, None, None, None, None, None)
+
+  implicit val eqSpectroscopyConfigurationOptions: Eq[SpectroscopyScienceRequirements] = Eq.by(x =>
+    (x.wavelength,
+     x.resolution,
+     x.signalToNoise,
+     x.signalToNoiseAt,
+     x.wavelengthRange,
+     x.focalPlane,
+     x.focalPlaneAngle,
+     x.capabilities
+    )
+  )
+}
+
+object SpectroscopyScienceRequirementsModel {
+  final case class Input(
+    wavelength:       Option[WavelengthModel.Input],
+    resolution:       Option[PosInt],
+    signalToNoise:    Option[PosBigDecimal],
+    signalToNoiseAt:  Option[WavelengthModel.Input],
+    wavelengthRange:  Option[WavelengthModel.Input],
+    focalPlane:       Option[FocalPlane],
+    focalPlaneAngle:  Option[FocalPlaneAngleInput],
+    capabilities:     Option[SpectroscopyCapabilities]
+  ) {
+    val create: ValidatedInput[SpectroscopyScienceRequirements] =
+      (wavelength.traverse(_.toWavelength("wavelength")),
+       signalToNoiseAt.traverse(_.toWavelength("signalToNoiseAt")),
+       wavelengthRange.traverse(_.toWavelength("wavelengthRange")),
+       focalPlaneAngle.traverse(_.toAngle)
+      ).mapN { (cw, signalToNoiseAt, wavelengthRange, focalPlaneAngle) =>
+        SpectroscopyScienceRequirements(cw, resolution, signalToNoise, signalToNoiseAt, wavelengthRange, focalPlane, focalPlaneAngle, capabilities)
+      }
+    val edit: ValidatedInput[SpectroscopyScienceRequirements] =
+      create
+  }
+
+  object Input {
+    val Default: Input = Input(None, None, None, None, None, None, None, None)
+
+    implicit val DecoderCreate: Decoder[Input] = deriveDecoder
+  }
+
+  final case class FocalPlaneAngleInput(
+    microarcseconds: Option[Long],
+    milliarcseconds: Option[BigDecimal],
+    arcseconds:      Option[BigDecimal]
+  ) {
+    import Units._
+
+    val toAngle: ValidatedInput[Angle] =
+      ValidatedInput.requireOne("focalPlaneAngle",
+        microarcseconds.map(Microarcseconds.readLong),
+        milliarcseconds.map(Milliarcseconds.readDecimal),
+        arcseconds     .map(Arcseconds.readDecimal)
```

Comment:
I didn't provide a `fromLong` and `fromDecimal` as it seems you can't really tell the units for decimals. Or do we want those methods?

Reply:
See, for example, `FiniteDurationModel.Input.fromLong` and `fromDecimal`.  You specify the units along with the value.  It allows the user to use long or decimal values regardless of the units.

That said, those may be overkill.  Maybe we should ask Andy whether he finds it at all useful.  If not, we should remove them everywhere.  If so, we should add it here.